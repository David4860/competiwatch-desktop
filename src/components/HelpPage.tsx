import React from 'react'
import { openLinkInBrowser } from "../utils/electronUtils";

interface Props {
  onPageChange: (activePage: string, val1?: any, val2?: any) => void;
}

const HelpPage = ({ onPageChange }: Props) => {
  const openLink = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
    event.preventDefault()
    const link = event.currentTarget
    openLinkInBrowser(link.href);
  };

  return (
    <div className="container layout-children-container">
      <div className="mt-4">
        <button
          type="button"
          onClick={() => onPageChange('accounts')}
          className="btn-link"
        >&larr; Back to your accounts</button>
      </div>
      <h1
        className="h1 mb-2 mt-4"
      >Help</h1>
      <div className="clearfix mb-4">
        <div className="col-md-6 float-left">
          <h2 className="h3 mb-2 text-normal">F.A.Q.</h2>
          <dl className="mr-4-md">
            <dt className="mb-1 text-bold">Does Competiwatch log my matches for me?</dt>
            <dd className="mb-4">
              No. You have to manually enter data after your games in order to see
              trends and other data about your competitive season. You might be interested
              in <a
                href="https://overtrack.gg/"
                onClick={openLink}
              >OverTrack</a> if you want automated tracking.
            </dd>
            <dt className="mb-1 text-bold">How much data do I have to enter when logging matches?</dt>
            <dd className="mb-4">
              <p>When logging placement matches, the result of the match
              (win/loss/draw) is required. For regular matches, only your new SR is required.</p>
              All other fields, like the map, heroes you played, and people you grouped
              with, are optional. However, if you don't fill in a field, you won't see trends
              about it. For example, if you never log the map you played on, you won't see a
              chart of win percentage by map.
            </dd>
            <dt className="mb-1 text-bold">Where do I get CSV files to import match data?</dt>
              <dd className="mb-4">
                If you've been logging your competitive matches in a spreadsheet with a tool
                like Excel or Google Sheets, you can produce a CSV (comma-separated value) file
                of the data you've already logged. Then you can import this CSV file into
                Competiwatch so you don't have to manually re-enter all those matches.
              </dd>
              <dt className="mb-1 text-bold">How much does it cost to use Competiwatch?</dt>
              <dd className="mb-4">
                It's free! <span role="img" aria-label="Tada">ðŸŽ‰</span> See <button
                  type="button"
                  className="btn-link"
                  onClick={() => onPageChange('about')}
                >the about page</button> for
                more information, but Competiwatch is my hobby project. Pull requests
                are welcome <a
                  href="https://github.com/cheshire137/competiwatch-desktop/"
                  onClick={openLink}
                >on GitHub</a> if you want to help out.
              </dd>
          </dl>
        </div>
        <div className="col-md-6 float-left">
          <h2 className="h3 mb-2 text-normal">Something not working?</h2>
          <p>
            <span>Please </span>
            <a
              href="https://github.com/cheshire137/competiwatch-desktop/issues"
              onClick={openLink}
            >open an issue</a>
            <span> on </span>
            GitHub with details about what isn't working
            in the app. Steps to reproduce the problem will be very helpful!
          </p>
        </div>
      </div>
    </div>
  );
};

export default HelpPage;
