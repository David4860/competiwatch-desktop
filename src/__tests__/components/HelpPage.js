import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import HelpPage from "../../components/HelpPage";

it("matches snapshot", () => {
  const tree = renderer.create(<HelpPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
