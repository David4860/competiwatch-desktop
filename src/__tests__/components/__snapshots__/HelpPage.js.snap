// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Box-sc-1b6inku-0 sc-AxjAm bAeCJN"
>
  <div
    className="mt-4"
  >
    <button
      className="ButtonBase-exshql-0 Button-sc-1cuu878-0 sc-AxirZ hiXGGL"
      onClick={[Function]}
      type="button"
    >
      ‚Üê Back to your accounts
    </button>
  </div>
  <h1
    className="h1 mb-2 mt-4"
  >
    Help
  </h1>
  <div
    className="clearfix mb-4"
  >
    <div
      className="col-md-6 float-left"
    >
      <h2
        className="h3 mb-2 text-normal"
      >
        F.A.Q.
      </h2>
      <dl
        className="mr-4-md"
      >
        <dt
          className="mb-1 text-bold"
        >
          Does Competiwatch log my matches for me?
        </dt>
        <dd
          className="mb-4"
        >
          No. You have to manually enter data after your games in order to see trends and other data about your competitive season. You might be interested in
           
          <a
            href="https://overtrack.gg/"
            onClick={[Function]}
          >
            OverTrack
          </a>
           
          if you want automated tracking.
        </dd>
        <dt
          className="mb-1 text-bold"
        >
          How much data do I have to enter when logging matches?
        </dt>
        <dd
          className="mb-4"
        >
          <p>
            When logging placement matches, the result of the match (win/loss/draw) is required. For regular matches, only your new SR is required.
          </p>
          All other fields, like the map, heroes you played, and people you grouped with, are optional. However, if you don't fill in a field, you won't see trends about it. For example, if you never log the map you played on, you won't see a chart of win percentage by map.
        </dd>
        <dt
          className="mb-1 text-bold"
        >
          Where do I get CSV files to import match data?
        </dt>
        <dd
          className="mb-4"
        >
          If you've been logging your competitive matches in a spreadsheet with a tool like Excel or Google Sheets, you can produce a CSV (comma-separated value) file of the data you've already logged. Then you can import this CSV file into Competiwatch so you don't have to manually re-enter all those matches.
        </dd>
        <dt
          className="mb-1 text-bold"
        >
          How much does it cost to use Competiwatch?
        </dt>
        <dd
          className="mb-4"
        >
          It's free!
           
          <span
            aria-label="Tada"
            role="img"
          >
            üéâ
          </span>
           
          See
           
          <button
            className="ButtonBase-exshql-0 Button-sc-1cuu878-0 sc-AxirZ hiXGGL"
            onClick={[Function]}
            type="button"
          >
            the about page
          </button>
           
          for more information, but Competiwatch is my hobby project. Pull requests are welcome
           
          <a
            href="https://github.com/cheshire137/competiwatch-desktop/"
            onClick={[Function]}
          >
            on GitHub
          </a>
           
          if you want to help out.
        </dd>
      </dl>
    </div>
    <div
      className="col-md-6 float-left"
    >
      <h2
        className="h3 mb-2 text-normal"
      >
        Something not working?
      </h2>
      <p>
        <span>
          Please 
        </span>
        <a
          href="https://github.com/cheshire137/competiwatch-desktop/issues"
          onClick={[Function]}
        >
          open an issue
        </a>
        <span>
           on 
        </span>
        GitHub with details about what isn't working in the app. Steps to reproduce the problem will be very helpful!
      </p>
    </div>
  </div>
</div>
`;
